File: src/main/java/toast/client/utils/TimerUtil.java
package toast.client.utils;

public class TimerUtil {
    private long lastMS;

    public TimerUtil() { setLastMS(); }

    public int convertToMS(int perSecond) { return 1000 / perSecond; }

    public boolean everyDelay(long delay) {
        if (System.currentTimeMillis() - this.lastMS >= delay) {
            this.lastMS = Math.min(System.currentTimeMillis(), this.lastMS + delay);
            return true;
        }
        return false;
    }

    public long getCurrentMS() { return System.nanoTime() / 1000000L; }

    public long getLastMS() { return this.lastMS; }

    public boolean hasReached(float f) { return ((float)(getCurrentMS() - this.lastMS) >= f); }

    public boolean isDelayComplete(long delay) {
        if (System.currentTimeMillis() - this.lastMS >= delay)
            return true;
        return false;
    }

    public int pastTime() { return (int)(System.currentTimeMillis() - this.lastMS); }

    public void reset() { this.lastMS = getCurrentMS(); }

    public void setLastMS() { this.lastMS = System.currentTimeMillis(); }

    public void setLastMS(long currentMS) { this.lastMS = currentMS; }
}

File: src/main/java/toast/client/utils/TimerUtil.kt
package toast.client.utils

/**
 * Class contains methods to time things and get delays
 */
class TimerUtil {
    private var lastMS: Long = 0

    fun everyDelay(delay: Long): Boolean {
        if (System.currentTimeMillis() - lastMS >= delay) {
            lastMS = System.currentTimeMillis().coerceAtMost(lastMS + delay)
            return true
        }
        return false
    }

    fun hasReached(f: Float): Boolean {
        return (System.currentTimeMillis() - lastMS).toFloat() >= f
    }

    /**
     * Checks if and amount of time has passed since the last reset
     */
    fun isDelayComplete(delay: Double): Boolean {
        return System.currentTimeMillis() - lastMS >= delay
    }

    /**
     * Returns the amount of time in milliseconds that has passed since last resetting the timer
     */
    fun pastTime(): Int {
        return (System.currentTimeMillis() - lastMS).toInt()
    }

    /**
     * Sets the last recorded millisecond to the current time in milliseconds
     */
    fun reset() {
        lastMS = System.currentTimeMillis()
    }

    init {
        reset()
    }
}